src/main/java/de/zalando/zmon/dataservice/data/MetricTiers.java

```diff
@@ -65,20 +85,22 @@ public void updateTiers() {
     }
 
     public boolean isMetricEnabled(int checkId) {
-        if (ingestMaxCheckTier <= 0 || ingestMaxCheckTier >= 3) return true;
-        else if (ingestMaxCheckTier == 1) return criticalChecks.contains(checkId);
-        else return criticalChecks.contains(checkId) || importantChecks.contains(checkId);
+        if (ingestMaxCheckTier == 1) {
+            return criticalChecks.contains(checkId);
+        } else if (ingestMaxCheckTier == 2) {
+            boolean isWhitelisted = criticalChecks.contains(checkId) || importantChecks.contains(checkId);
+            boolean isSampled = sampledCheckTier != 2 || random.nextDouble() > sampledCheckRate;
```

Comment:
agree, but `>2` is also not working good, since it is then `true` for 3 and the effect is not what we want. This one is better compromise

Reply:
Despite it is merged, I would like to clarify, if you don't mind.
In case of `sampledCheckTier == 3` both `sampledCheckTier != 2` and `sampledCheckTier > 2` will be true.